bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/channelhelper/CameraChannelHelper.java

```diff
@@ -36,64 +37,72 @@
-                case CHANNEL_LIVEPICTURE_LOCAL_URL:
-                    return toStringType(getLivePictureURL(true, isMonitoring));
-                case CHANNEL_LIVEPICTURE:
-                    return toRawType(getLivePictureURL(isLocal, isMonitoring));
-                case CHANNEL_LIVESTREAM_VPN_URL:
-                    return getLiveStreamURL(false, (String) config.get(QUALITY_CONF_ENTRY), isMonitoring);
-                case CHANNEL_LIVESTREAM_LOCAL_URL:
-                    return getLiveStreamURL(true, (String) config.get(QUALITY_CONF_ENTRY), isMonitoring);
-            }
+            return switch (channelId) {
+                case CHANNEL_MONITORING -> camera.getMonitoring();
+                case CHANNEL_SD_CARD -> toStringType(camera.getSdStatus());
+                case CHANNEL_ALIM_STATUS -> toStringType(camera.getAlimStatus());
+                default -> liveChannels(channelId, (String) config.get(QUALITY_CONF_ENTRY), camera,
+                        OnOffType.ON.equals(camera.getMonitoring()), localUrl != null);
+            };
         }
         return null;
     }
 
-    private @Nullable String getLivePictureURL(boolean local, boolean isMonitoring) {
-        String url = local ? localUrl : vpnUrl;
-        if (!isMonitoring || (local && !isLocal) || url == null) {
+    public @Nullable String getLivePictureURL(boolean local, boolean isMonitoring) {
+        if (!isMonitoring || (local && (localUrl != null))) {
             return null;
         }
-        return "%s%s".formatted(url, LIVE_PICTURE);
+        return "%s%s".formatted(getUrl(local), LIVE_PICTURE);
+    }
+
+    private @Nullable State liveChannels(String channelId, String qualityConf, HomeStatusModule camera,
+            boolean isMonitoring, boolean isLocal) {
+        if (vpnUrl == null) {
+            setUrls(Objects.requireNonNull(camera.getVpnUrl()), localUrl);
+        }
+        return switch (channelId) {
+            case CHANNEL_LIVESTREAM_LOCAL_URL ->
+                isLocal ? getLiveStreamURL(true, qualityConf, isMonitoring) : UnDefType.NULL;
+            case CHANNEL_LIVEPICTURE_LOCAL_URL ->
+                isLocal ? toStringType(getLivePictureURL(true, isMonitoring)) : UnDefType.NULL;
+            case CHANNEL_LIVESTREAM_VPN_URL -> getLiveStreamURL(false, qualityConf, isMonitoring);
+            case CHANNEL_LIVEPICTURE_VPN_URL -> toStringType(getLivePictureURL(false, isMonitoring));
+            case CHANNEL_LIVEPICTURE -> {
+                State result = toRawType(getLivePictureURL(isLocal, isMonitoring));
+                if (UnDefType.NULL.equals(result) && isLocal) {// If local read is unsuccessfull, try the VPN version
+                    result = toRawType(getLivePictureURL(false, isMonitoring));
+                }
+                yield result;
+            }
+            default -> null;
+        };
+    }
+
+    private String getUrl(boolean local) {
+        return Objects.requireNonNull(local ? localUrl : vpnUrl);
```

Comment:
`localUrl ` can be null, no ?
It is declared with `@Nullable`

Reply:
Changed